adapters/handlers/rest/handlers_nodes.go

```diff
@@ -33,36 +34,55 @@ type nodesHandlers struct {
 	metricRequestsTotal restApiRequestsTotal
 }
 
-func (s *nodesHandlers) getNodesStatus(params nodes.NodesGetParams, principal *models.Principal) middleware.Responder {
-	nodeStatuses, err := s.manager.GetNodeStatus(params.HTTPRequest.Context(), principal, "")
+func (n *nodesHandlers) getNodesStatus(params nodes.NodesGetParams, principal *models.Principal) middleware.Responder {
+	output, err := verbosity.ParseOutput(params.Output)
 	if err != nil {
-		return s.handleGetNodesError(err)
+		return nodes.NewNodesGetUnprocessableEntity().WithPayload(errPayloadFromSingleErr(err))
+	}
+
+	nodeStatuses, err := n.manager.GetNodeStatus(params.HTTPRequest.Context(), principal, "", output)
+	if err != nil {
+		return n.handleGetNodesError(err)
 	}
 
 	status := &models.NodesStatusResponse{
 		Nodes: nodeStatuses,
 	}
 
-	s.metricRequestsTotal.logOk("")
-	return nodes.NewNodesGetOK().WithPayload(status)
+	n.metricRequestsTotal.logOk("")
+	return nodes.NewNodesGetOK().WithPayload(n.nodesStatusResponse(status, output))
 }
 
-func (s *nodesHandlers) getNodesStatusByClass(params nodes.NodesGetClassParams, principal *models.Principal) middleware.Responder {
-	nodeStatuses, err := s.manager.GetNodeStatus(params.HTTPRequest.Context(), principal, params.ClassName)
+func (n *nodesHandlers) getNodesStatusByClass(params nodes.NodesGetClassParams, principal *models.Principal) middleware.Responder {
+	output, err := verbosity.ParseOutput(params.Output)
 	if err != nil {
-		return s.handleGetNodesError(err)
+		return nodes.NewNodesGetUnprocessableEntity().WithPayload(errPayloadFromSingleErr(err))
+	}
+
+	nodeStatuses, err := n.manager.GetNodeStatus(params.HTTPRequest.Context(), principal, params.ClassName, output)
+	if err != nil {
+		return n.handleGetNodesError(err)
 	}
 
 	status := &models.NodesStatusResponse{
 		Nodes: nodeStatuses,
 	}
 
-	s.metricRequestsTotal.logOk("")
-	return nodes.NewNodesGetOK().WithPayload(status)
+	n.metricRequestsTotal.logOk("")
+	return nodes.NewNodesGetOK().WithPayload(n.nodesStatusResponse(status, output))
+}
+
+func (n *nodesHandlers) nodesStatusResponse(resp *models.NodesStatusResponse, output string) *models.NodesStatusResponse {
+	if output == verbosity.OutputMinimal {
+		for _, node := range resp.Nodes {
+			node.Shards = nil
```

Comment:
hmm, I can see that you are making `Shards` `nil`, how would it be complicated to pass the `verbosity` param  to method that gets the nodes information so that `Shards` information would not be retrieved?

Reply:
Yeah this is something I had considered during implementation... I looked at the implementation for batch delete, and it is essentially doing this same thing. I like this approach because conceptually, the request details are not being leaked into the usecases layer. However, it could be more efficient to not assign to `shards` in the first place. What do you think?

Reply:
```
"stats": {
	"objectCount": 6,
	"shardCount": 1
},
```
`stats` need looping through all shards anyway, so it is not that we can skip shards entirely by passing `verbosity` param deeper.
Current solution seems to be good enough for me

Reply:
then let's merge this

Reply:
Ok, done

Reply:
I made the change, because I noticed that if there are many shards, we can skip the overhead of locking each shard to get the queue length, indexing status, etc if the output requested is `minimal`

Reply:
So now shardCount is calculated on the fly, rather than relying on the `len(shards)`